### DISCUSSION TITLE
Dark Library

### DISCUSSION THREAD

**Discussion by fly800** [aaronwmorris/indi-allsky#1552] [Q&A] [answered]
Hello,

Is there a way to copy dark frames from one camera to another?  No sure why, I have two cameras to select from but they were always the same camera.    The two selectable cameras are:

1. ZWO CCD ASI120MM Mini
2. ZWO ASI120MM Mini

Greatly appreciate a solution, but I really don't want restart the Dark Library process again.

**Comment by aaronwmorris** [comment, marked as answer]
With new releases of INDI, sometimes the camera names change.  With ZWO, the name might change if you switch between `indi_asi_ccd` and `indi_asi_single_ccd`.

Copying the darks to a new camera is possible.  Two possibilities:

1. Maintain darks for both cameras:  You just need to copy the FITS files in `/var/www/html/allsky/images/darks/` to new file names.  The filenames do not really matter too much, just that they end with `.fit` or `.fits`.  Once you copy the files, you can run `./misc/import_darks_frames.py` to import the frames for the new camera name.
2. Move the darks to the new camera.  You just need to know the camera ID of the old and new camera.  It is probably 1 and 2, respectively.  Run the sql:

        sqlite3 /var/lib/indi-allsky/indi-allsky.sqlite "UPDATE darkframe SET camera_id=2 WHERE camera_id=1;"
        
        sqlite3 /var/lib/indi-allsky/indi-allsky.sqlite "UPDATE badpixelmap SET camera_id=2 WHERE camera_id=1;"

You can find the camera IDs with the cameras view:  https://hostname/indi-allsky/cameras

**Reply by fly800** [reply]
Great thanks.  Works like a charm.
